File: examples/exp-ticket-support/functions/ticket_classify.py
import json


def new():
    return Function()


class Function:
    async def handle(self, scope, receive, send):
        message = await receive()
        body = message.get("body", b"")
        event = json.loads(body.decode("utf-8")) if body.strip() else {}
        text = str(event.get("text", ""))
        lower = text.lower()

        if any(word in lower for word in ["refund", "money", "charge"]):
            category = "refund"
        elif any(word in lower for word in ["password", "login", "error", "bug"]):
            category = "technical"
        elif any(word in lower for word in ["angry", "complaint", "terrible"]):
            category = "complaint"
        else:
            category = "general"

        result = {
            **event,
            "category": category,
            "classified_by": scope["minik8s"]["name"],
        }
        await respond(send, result)


async def respond(send, result):
    payload = json.dumps(result).encode("utf-8")
    await send({
        "type": "http.response.start",
        "status": 200,
        "headers": [[b"content-type", b"application/json"]],
    })
    await send({"type": "http.response.body", "body": payload})

File: examples/exp-ticket-support/functions/ticket_classify.rs
use crate::FunctionContext;
use serde_json::{Map, Value};

pub fn handle(event: Value, ctx: &FunctionContext) -> Value {
    let text = event
        .get("text")
        .and_then(Value::as_str)
        .unwrap_or("")
        .to_ascii_lowercase();
    let category = if contains_any(&text, &["refund", "money", "charge"]) {
        "refund"
    } else if contains_any(&text, &["password", "login", "error", "bug"]) {
        "technical"
    } else if contains_any(&text, &["angry", "complaint", "terrible"]) {
        "complaint"
    } else {
        "general"
    };

    let mut result = object_or_empty(event);
    result.insert("category".to_string(), Value::String(category.to_string()));
    result.insert("classified_by".to_string(), Value::String(ctx.name.clone()));
    Value::Object(result)
}

fn contains_any(text: &str, words: &[&str]) -> bool {
    words.iter().any(|word| text.contains(word))
}

fn object_or_empty(value: Value) -> Map<String, Value> {
    match value {
        Value::Object(object) => object,
        _ => Map::new(),
    }
}
